app/helpers/container_summary_helper.rb

```diff
@@ -98,9 +98,15 @@ def textual_collection_link(collection, as: nil, feature: nil)
 
     if count > 0 && role_allows(:feature => feature)
       if collection.respond_to?(:proxy_association)
-        h[:link] = url_for(:action  => 'show',
-                           :id      => collection.proxy_association.owner,
-                           :display => collection.proxy_association.reflection.name)
+        if collection.count == 1
+          h[:link] = url_for(:controller => klass.name.underscore,
+                             :action     => 'show',
+                             :id         => collection.first)
```

Comment:
the title for the navigation should be set differently for a single item vs. collection
'Show all X' versus 'Show X'
(in other words, the statement  h[:title] = "Show all #{label}" from couple of lines below should be customized)